src/components/CardListBox/CardListBox.tsx

```diff
@@ -0,0 +1,101 @@
+import { ChangeEvent, useEffect, useState } from "react";
+import CardSearchInput from "./CardSearchInput/CardSearchInput.tsx";
+import useGet from "../../hooks/useGet";
+import CardFolderList from "./CardFolderList/CardFolderList.tsx";
+import styles from "./CardListBox.module.scss";
+import CardList from "../CardList/CardList.tsx";
+import { END_POINT } from "../../constants/index.ts";
+import getFormattedLinks from "@/hooks/getFormattedLinks";
+import LinksEmptyCase from "./LinksEmptyCase/LinksEmptyCase.tsx";
+import { FormetLink, ObjectLink, ObjectFolder } from "@/constants/index.types";
+
+export default function CardListBox() {
+  // 북마크 Folders 상태
+  const [folders, setFolders] = useState<ObjectFolder[]>([]);
+  // 렌더링되는 Card - Links 상태
+  const [links, setLinks] = useState<ObjectLink[]>([]);
+  // 북마크 Folder의 id로 END_POINT.links + query 변경
+  const [changeLinksFolder, setChangeLinksFolder] = useState(END_POINT.links);
+  // select한 북마크 Folder의 이름으로 title 변경
+  const [linksTitle, setLinksTitle] = useState(`전체`);
+  // 전체 데이터 배열 저장 = 검색용
+  const [searchLinks, setSearchLinks] = useState<ObjectLink[]>([]);
+  // inputValue 상태
+  const [searchValue, setSearchValue] = useState("");
+
+  const { data: foldersData, isLoading: isFoldersLoading } = useGet<
+    ObjectFolder[]
+  >(END_POINT.folders);
+
+  const { data: linksData, isLoading: isLinksLoading } =
+    useGet<FormetLink[]>(changeLinksFolder);
+
+  useEffect(() => {
+    if (!isFoldersLoading && !isLinksLoading) {
+      setFolders(foldersData);
+      const strangeData = linksData;
+      const rightData = getFormattedLinks(strangeData);
+      setLinks(rightData);
+      setSearchLinks(rightData);
+    }
+  }, [foldersData, isFoldersLoading, linksData, isLinksLoading]);
+
+  const handleFolderClick = (folder: ObjectFolder) => {
+    setChangeLinksFolder(`${END_POINT.links}?folderId=${folder.id}`);
+    setLinksTitle(folder.name);
+  };
+
+  const handleTotalBtnClick = () => {
+    setChangeLinksFolder(END_POINT.links);
+    setLinksTitle(`전체`);
+  };
+
+  const handleSearchInputCancel = () => {
+    setSearchValue("");
+    setLinks(searchLinks);
+  };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value);
+  };
+
+  useEffect(() => {
+    if (searchValue !== "") {
+      const filteredLinks = searchLinks.filter((link) => {
+        return Object.values(link)
+          .join("")
+          .toLowerCase()
+          .includes(searchValue.toLowerCase());
+      });
+      setLinks(filteredLinks);
+    } else {
+      setLinks(searchLinks);
+    }
+  }, [searchValue]);
```

Comment:
요것도 useEffect 내부보다는, 외부에서 해주는게 더 괜찮을 것 같구요!
(useEffect 사용은 최대한 지양해보면 좋을 것 같아요)

그리고 조건문도 else문 사용은 최대한 지양하고, 요기서는 early return으로 이렇게 처리해주면 좋을 것 같습니다!

```suggestion
  useEffect(() => {
    if (searchValue === "") {
      setLinks(searchLinks);
      return;
    }
    
    const filteredLinks = searchLinks.filter((link) => {
      return Object.values(link)
        .join("")
        .toLowerCase()
        .includes(searchValue.toLowerCase());
    });
    setLinks(filteredLinks);
  }, [searchValue]);
```

Reply:
멘토님 이부분 계속해서 inputValue가 변경될때 마다 렌더링이 일어나게 되어있는데
setTimeout과 같은 구조가 필요한지 궁금합니다 보통 검색 기능에서는 어떤 구조로 설계가 이루어지는지도 알고싶습니다

Reply:
우선 리렌더링이 발생하는 것 자체를 단점으로 보는건 좋지 않구요!
왜냐면 리액트는 원래 그렇게 동작하는게 당연하니까요!

말씀주신것처럼 검색같은 기능을 구현할때는 요구사항에 따라 다르지만, 보통은 디바운스(내부적으로는 setTimeout 사용)를 적용하는 편이에요. 디바운스라는 기법을 이용하면 값을 입력하면 바로 상태값에 반영하는것이 아니라, 특정 시간 이후에 상태값에 반영하도록 딜레이를 줄 수 있어요. 그래서 값을 입력을 해도 바로 렌더링을 발생시키지 않고 딜레이 이후에 렌더링을 발생시킵니다.
(별도로 쓰로틀링에 대해서도 알아봐주시면 좋습니다!)

딜레이 시간은 0.2~0.3초 정도로 보통 많이 사용하는 것 같구요! 서비스에 따라서 따로 디바운스를 걸어놓지 않은 곳도 있어요!